Homework2/LazyTest/LazyMultiThread.Test.cs

```diff
@@ -0,0 +1,34 @@
+﻿namespace Lazy.Test;
+
+public class LazyMultiThreadTest
+{
+    [Test]
+    public void LazyShouldReturnSameResultsFromDifferentThreadsTest()
+    {
+        int sum = 0;
+        Func<int> function = () => { for (int i = 0; i < 10000; ++i) { sum += 1; }; return sum; };
+        var lazy = new LazyMultiThread<int>(function);
+        var threads = new Thread[10];
+        var resultInThreads = new int[threads.Length];
+        for (int i = 0; i < threads.Length; ++i)
+        {
+            var locali = i;
+            threads[locali] = new Thread(() => resultInThreads[locali] = lazy.Get());
```

Comment:
```suggestion
            threads[i] = new Thread(() => resultInThreads[locali] = lazy.Get());
```
Здесь идеологически правильней просто i использовать, проблем с замыканием ведь нет